### PR TITLE
Sof 463

### PR DESCRIPTION
**Pull request by hannemennah**

### CODE REVIEW

File: common/messages/dbstate.go [hunk 1]
```diff
@@ -185,9 +185,36 @@ func (m *DBStateMsg) Validate(state interfaces.IState) int {
...
+	validSigCount := 0
+
+	data, err := m.DirectoryBlock.GetHeader().MarshalBinary()
+	if err != nil {
+		state.AddStatus(fmt.Sprint("Debug: DBState Signature Error, Marshal binary errored"))
+		return validSigCount
+	}
+
+	for _, sig := range m.SignatureList.List {
+		check, err := state.VerifyAuthoritySignature(data, sig.GetSignature(), dbheight)
```
Comment (PaulSnow) [inline common/messages/dbstate.go]: How do we tell the difference between audit and federated servers here?

File: state/state.go [hunk 2]
```diff
@@ -926,7 +927,33 @@ func (s *State) LoadDBState(dbheight uint32) (interfaces.IMsg, error) {
 		panic(fmt.Sprintf("The configured network ID (%x) differs from the one in the local database (%x) at height %d", configuredID, dbaseID, dbheight))
 	}
 
-	msg := messages.NewDBStateMsg(s.GetTimestamp(), dblk, ablk, fblk, ecblk, eBlocks, entries)
+	blockSig := new(primitives.Signature)
+	var allSigs []interfaces.IFullSignature
+
+	nextABlock, err := s.DB.FetchABlockByHeight(dbheight + 1)
+	if err != nil || nextABlock == nil {
+		return nil, err
```
Comment (PaulSnow) [inline state/state.go]: This is where you can look in the process list to see if you can get the signatures there.

**Comment by hannemennah** [comment]
Updated to also check processLists for the signatures.

File: state/state.go [hunk 3]
```diff
@@ -932,27 +932,36 @@ func (s *State) LoadDBState(dbheight uint32) (interfaces.IMsg, error) {
...
 	nextABlock, err := s.DB.FetchABlockByHeight(dbheight + 1)
 	if err != nil || nextABlock == nil {
-		return nil, err
-	}
-	abEntries := nextABlock.GetABEntries()
-	for _, adminEntry := range abEntries {
-		data, err := adminEntry.MarshalBinary()
-		if err != nil {
-			return nil, err
+		if dbheight < s.ProcessLists.DBHeightBase || int(dbheight) > int(s.ProcessLists.DBHeightBase)+len(s.ProcessLists.Lists) {
```
Comment (PaulSnow) [inline state/state.go]: This test isn't necessary.  You can just get the process list, and if it is nil, then you don't have one.

**Pull request merged**